src/reducers/saveMarkdownReducer.js

```diff
@@ -0,0 +1,27 @@
+import { TAB_NAME_CHANGE, UPDATE_HISTORY, NEW_HISTORY } from '../actions/saveMarkdownActions';
+
+
+const initialState = {
+  tabName: '',
```

Comment:
is current tab in your document reducer tracking the same thing as 'tabName' here?